server/task_executor_schema_update_ghost_sync.go

```diff
@@ -237,11 +237,11 @@ func (exec *SchemaUpdateGhostSyncTaskExecutor) runGhostMigration(ctx context.Con
 		return true, nil, errors.Wrap(err, "failed to init migrationContext for gh-ost")
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
 	migrator := logic.NewMigrator(migrationContext, "bb")
 
-	go func(ctx context.Context) {
+	ctx1, cancel := context.WithCancel(ctx)
```

Comment:
s/ctx1/childCtx where childCtx is more readable.

Also, is there any reason that we need a cancel() here? since the cancellation of ctx is done by the parent cancel().

Reply:
Renamed.

The cancel() is needed here. It cancels the go routine when the parent function returns on complete or error.